### ISSUE TITLE
Run docker builds while waiting on resource dependencies

### ISSUE THREAD

**Issue by rpocase** [windmilleng/tilt#2770]
I have a Tiltfile that contains three docker builds. Each of these is associated with a `k8s_resource` that has at least one dependency. The upstream resources take quite a long time for initial setup (~10 minutes). This setup is not particularly process intensive, so I would like to be able to start the docker build processes while the dependent resources are set up.

Combined with the just released parallel build feature, this has the potential to reduce clean slate upstart by around 5-10 minutes.

**Comment by andymartin-sch** [comment]
+1, I noticed the same thing after we started using the new `max_parallel_updates`

**Comment by nicks** [comment]
Yes! Thanks for filing. We're collecting feedback on this post-launch, and this might be one of the first follow-ups we do.

**Comment by nicks** [comment]
We're doing a major rewrite of Tilt as an apiserver that reconciles Image builds very differently, so I'm hoping this will get fixed when we move to that system. (We've already got Kubernetes applys on that system - https://api.tilt.dev/kubernetes/kubernetes-apply-v1alpha1.html - and Image builds is the next big one)

**Comment by Meemaw** [comment]
Any update on this one @nicks ? This would be huge!

**Comment by nicgrayson** [comment]
@nicks Do you have any updates on this? This is blocking me from using Tilt in CI because Tilt takes so much longer
